agent/local/state.go

```diff
@@ -1251,9 +1346,14 @@ func (l *State) syncCheck(key structs.CheckID) error {
 
 	var out struct{}
 	err := l.Delegate.RPC("Catalog.Register", &req, &out)
+	l.Lock()
+	defer l.Unlock()
 	switch {
 	case err == nil:
 		l.checks[key].InSync = true
```

Comment:
We need to check if the key is in the map here as well.

```suggestion
                if c, ok := l.checks[key]; ok && c.version == currentVersion {
                  c.InSync = true
                }
```